marketing/sailthruScriptJob.groovy

```diff
@@ -0,0 +1,130 @@
+package devops
+
+import org.yaml.snakeyaml.Yaml
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_GITHUB_BASEURL
+
+
+/*
+Example secret YAML file used by this script
+
+    oauth_host : The host used to obtain Course Discovery access token
+    oauth_key : Key used to obtain Course Discovery access token
+    oauth_secret : Secret used to obtain Course Discovery access token
+    sailthru_key : Access key for Sailthru api
+    sailthru_secret : Access secret for Sailthru api
+    content_api_url : Url of Course Discovery API
+    lms_url : Url of LMS (default http://courses.edx.org
+    report_email : Email address to sent batch report to
+    type : A choice of "programs", "courses" or "all" 
+    fixups : any fix
+    mode : command mode
+    requirements : path to sailthru requirements file in repo
+    scriptPath : the script to run for this job
+    githubRepoUrlSegment : url segment of repo i.e edx/repo_name
+    jobName : job-name-to-be-created
+    notificationEmail : Post Action to send email for job notification 
+*/
+
+/* stdout logger */
+/* use this instead of println, because you can pass it into closures or other scripts. */
+Map config = [:]
+Binding bindings = getBinding()
+config.putAll(bindings.getVariables())
+PrintStream out = config['out']
+
+/* Map to hold the k:v pairs parsed from the secret file */
+Map secretMap = [:]
+out.println('Parsing secret YAML file')
+try {
+    /* Parse k:v pairs from the secret file referenced by secretFileVariable */
+    String contents = new File("${EDX_SAILTHRU_SYNC_CONTENT}").text
```

Comment:
Pull these secrets from the ecom-secure repo instead of a file on disk. Follow the pattern established here: https://github.com/edx-ops/edx-jenkins-job-dsl/blob/master/jobs/tools-edx-jenkins.edx.org/createTranslationJobs.groovy#L40